include/range/v3/algorithm/shuffle.hpp

```diff
@@ -76,6 +76,16 @@ namespace ranges
                 return orig;
             }
 
+            template<typename I, typename S,
+                CONCEPT_REQUIRES_(RandomAccessIterator<I>() && IteratorRange<I, S>() &&
+                    Permutable<I>())>
+            I operator()(I begin, S end_) const
+            {
+                std::random_device rng;
+                std::mt19937 urng(rng());
```

Comment:
Do we want `urng` to be `thread_local` to avoid initialization cost on repeat calls in the same thread? I think yes. Do we want to seed with more than a single `unsigned int` of entropy? I think no. In both cases, my answers are chosen to most closely resemble `random_shuffle` (i.e., I'm "weakly for").